### PR TITLE
Added Calendar and Movie Entities

### PR DESCRIPTION
**Pull request by CristutiuManu**
I added Calendar and Movie entities.

### CODE REVIEW

File: backend/Models/CalendarContext.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore;
```
Comment (danbris) [inline backend/Models/CalendarContext.cs:1]: Why is there a CalendarContext? Also a MovieCOntext?
Will you add a separate context for each entity????
https://docs.microsoft.com/en-us/ef/ef6/fundamentals/working-with-dbcontext
"The recommended way to work with context is to define a class that derives from DbContext and exposes DbSet properties that represent collections of the specified entities in the context"

File: backend/Startup.cs [hunk 2]
```diff
@@ -24,6 +27,9 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllersWithViews();
+            services.AddDbContext<MovieContext>(builder => builder.UseSqlServer("DevConnection"));
```
Comment (danbris) [inline backend/Startup.cs:30]: why are there so many contexts?? what's the idea?

File: backend/Models/Movie.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public DateTime ReleasedDate { get; set; }
+        public float Rating { get; set; }
+        public TimeSpan Duration { get; set; }
+        public Movie()
+        { }
+        public Movie(string T, string D, DateTime RD, float R, TimeSpan Dur)
+        {
+            ID = counter++;
```
Comment (danbris) [inline backend/Models/Movie.cs:22]: The Id(primary key) should be handled by EF. Imagine I send you a model with a negative ID, or with an invalid integer ID...

File: backend/Startup.cs (hunk 2 above)
Comment (CristutiuManu) [reply backend/Startup.cs:30]: I tried to generate a Controller for Calendar, but I gave up on the idea and I forgot to remove the DbContext for Calendar, sorry.

File: backend/Models/CalendarContext.cs (hunk 1 above)
Comment (CristutiuManu) [reply backend/Models/CalendarContext.cs:1]: I'll fix it asap

File: backend/Controllers/MovieController.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                return NotFound();
+            }
+
+            return movie;
+        }
+
+        
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutMovie(int id, Movie movie)
```
Comment (danbris) [inline backend/Controllers/MovieController.cs:46]: I had the same issue with a previous PR: why are you using db models directly and not DTOs? If you are using models, I can send you whatever data, without any restriction. Why should I care about movie.ID?

File: backend/Controllers/MovieController.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            return NoContent();
+        }
+
+        // POST: api/Movie
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for
+        // more details, see https://go.microsoft.com/fwlink/?linkid=2123754.
+        [HttpPost]
+        public async Task<ActionResult<Movie>> PostMovie(Movie movie)
+        {
+            _context.movies.Add(movie);
```
Comment (danbris) [inline backend/Controllers/MovieController.cs:80]: See my previous comment. I can send you here a model with a negative ID value, or with an existing ID value - you are directly saving it.

File: backend/Controllers/MovieController.cs (hunk 4 above)
Comment (CristutiuManu) [reply backend/Controllers/MovieController.cs:46]: I'm gonna fix that too.

File: backend/Models/Calendar.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CinemaApp.Models
+{
+    
+    public class Calendar
+    {
```
Comment (danbris) [inline backend/Models/Calendar.cs:12]: Shouldn't this model have an ID or something?

File: backend/Controllers/MovieController.cs (hunk 5 above)
Comment (CristutiuManu) [reply backend/Controllers/MovieController.cs:80]: I already talked with my colleague who made the Movie class about this problem. He will solve it.

File: backend/Models/Calendar.cs (hunk 6 above)
Comment (CristutiuManu) [reply backend/Models/Calendar.cs:12]: image

Comment (CristutiuManu) [reply backend/Models/Calendar.cs:12]: I just followed the schema :)

**Review by danbris — APPROVED** [review]
Approved just to unblock, but there are still issues to solve.

**Pull request merged**